### ISSUE TITLE
support weather.com api as an alternative to yahoo

### ISSUE THREAD

**Issue by rahulsundaram** [simon04/gnome-shell-extension-weather#120]
for my region, weather.com is far more reliable than Yahoo is.  Would you kindly add support for it?  Can be a configuration preference.

**Comment by rahulsundaram** [comment]
apparently weather.com uses the same data as http://graphical.weather.gov/xml/

**Comment by simon04** [comment]
This repository is no longer actively maintained. Please take a look at Neroth's fork and file a ticket there: https://github.com/Neroth/gnome-shell-extension-weather

**Comment by ferdnyc** [comment]
While I realize this repository is the wrong place for development of this
feature anyway, I just had to register my surprise regarding the initial
request. Not that I'm questioning or disagreeing with it... I guess I'm
just confused, as it was my impression that yahoo's data IS the
weather.com data. Or that "Yahoo Weather" was just a shell around it,
anyway.  

I mean, their Hourly and Detail forecast links — actually, pretty much all
of the links — on the Yahoo Weather page even take you to the corresponding
sections of weather.com. So, I'm surprised to hear that there would be any
discrepancy between the two.  

I wonder (and this is just me "thinking out loud", now) if the issue might
be one of update lag? Right now, the Yahoo Weather page for my region is
showing "Last Updated 09:49 EDT"... and it's currently 11:26 AM. So, it
could just be that Yahoo is refreshing too infrequently from their own
weather.com feed. That's disappointing, if it is indeed the issue.  

The barrier, from what I understand, to supporting
weather.com (specifically) as a data provider is that they really don't
seem to "get" open-source. Not when it comes to their data-consumption policies, anyway.
Their data access requirements, at least as developers who've wanted to
implement data clients have interpreted them (that I've seen), end up
forcing each person who compiles and builds the tool to register for a
separate API key, which is a big giant hassle for anyone who just wants to
compile a stupid weather tool. Not to mention, the weather extension ISN'T
compiled, and I don't believe it's permitted to distribute source
containing their API key — which makes me think they don't really "get"
interpreted languages, either.  

I don't know, maybe things have changed, and they've finally gotten their
heads screwed on straight about this stuff. Or, maybe the previous
interpretations of their policies were overly pessimistic, and the reality
isn't really as tedious & overly-complicated as it was thought to be from
the initial investigations. But, like I said, weather.com haven't appeared
to be the most welcoming resource, when choosing data providers for a
non-commercial, collaboratively-developed open-source project.

**Comment by ferdnyc** [comment]
Yeah, see... it's actually a frequency issue, I was misremembering slightly.

Reading here:

http://www.wunderground.com/weather/api/d/pricing.html

(which is where weather.com's "WEATHER™ API" link takes you to, for data
feed service), a free-as-in-beer API key is restricted to a _maximum_ of
500 requests per day, and a rate of no more than 10 requests per minute...
which, for any open-source project, is just unworkable. There's no way any
developer can promise to keep data consumption below that level. So to use
weather.com/wunderground as the data provider, each USER ends up having to
register for and configure their own API key. Big giant hassle.

(Plus, I'm not sure the data would be any better — the weather.org site
documents that their own REST feed is updated no more than once per hour.
Weather.com's own site typically has up-to-the-minute data, but their data
consumers don't necessarily get access to that same freshness... not
without shelling out lots of actual cash, anyway.
